features/netsocket/nsapi_dns.cpp

```diff
@@ -305,3 +519,337 @@ nsapi_error_t nsapi_dns_query(NetworkStack *stack, const char *host,
     address->set_addr(addr);
     return (nsapi_error_t)((result > 0) ? 0 : result);
 }
+
+nsapi_error_t nsapi_dns_query_async(NetworkStack *stack, const char *host,
+        NetworkStack::hostbyname_cb_t callback, void *data, nsapi_version_t version)
+{
+    nsapi_size_or_error_t result = nsapi_dns_query_multiple_async(stack, host, callback, data, 1, version);
+    return (nsapi_error_t)((result > 0) ? 0 : result);
+}
+
+static nsapi_error_t nsapi_dns_call_default(mbed::Callback<void()> func)
+{
+    events::EventQueue *event_queue = mbed::mbed_event_queue();
+    if (!event_queue) {
+        return NSAPI_ERROR_NO_MEMORY;
+    }
+    if (event_queue->call(func) == 0) {
+        return NSAPI_ERROR_NO_MEMORY;
+    }
+    return NSAPI_ERROR_OK ;
+}
+
+static nsapi_error_t nsapi_dns_call_in_default(int delay, mbed::Callback<void()> func)
```

Comment:
Suggestion - these become the base implementation of `NetworkStack::call`, and then the logic below becomes `if (dns_call) { custom} else { stack->call }`.